loki/transformations/tests/test_loop_blocking.py

```diff
@@ -0,0 +1,95 @@
+# (C) Copyright 2018- ECMWF.
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+
+import itertools
+import pytest
+import numpy as np
+from fparser.two.pattern_tools import subroutine
+
+from loki import available_frontends, Subroutine, pragmas_attached, find_driver_loops, Loop, fgen, \
+    pprint
+from loki.transformations.loop_blocking import block_loop
+
+
+@pytest.mark.parametrize('frontend', available_frontends())
+def test_simple_blocking(tmp_path, frontend):
+    """
+    Apply loop blocking of simple loops into two loops
+    """
+    fcode = """
+subroutine simple_loop(a, b, n)
+  integer, intent(in) :: n
+  real, intent(out) :: a(n)
+  real, intent(in) :: b(n,n)
+  real :: c(n)
+  integer :: i
+  !$loki driver-loop
+  do i=1,n
+    a(i) = i
+    c(1) = a(i)
+    call some_kernel(a(i), b(:,i))
+    b(:,i) = a(i)
+  end do
+end subroutine simple_loop
+    """
+    routine = Subroutine.from_source(fcode, frontend=frontend)
+
+    # loops = FindNodes(ir.Loop).visit(routine.ir)
+    with pragmas_attached(routine, Loop):
+        loops = find_driver_loops(routine,
+                                  targets=None)
+
+    for loop in loops:
+        block_loop(routine, loop, 10)
+
+    print()
+    print(30 * "-" + "IR after manipulations" + 30 * "-")
+    print(f"{pprint(routine.ir)}")  # scheduler['transformation_module_hoist#driver'].ir.ir)}")
+    print()
+    print(30 * "-" + "Generated FORTRAN" + 30 * "-")
+    print(f"{fgen(routine)}")  # scheduler['transformation_module_hoist#driver'].ir.ir)}")
```

Comment:
This is a good start when developing the utility. For a full test we would need to replace this by introspection of the transformed IR to ensure we can programmatically detect that the produced code matches the expectations.
Since this is a neutral modification of the control flow, you could even use the JIT mechanism to compile and run original as well as transformed code and compare the result. See the following test as an example: https://github.com/ecmwf-ifs/loki/blob/9a76c212914780570585e5ac34bc52ef69cb60f5/loki/transformations/tests/test_transform_loop.py#L29